src/app/pages/psbt-request/components/psbt-decoded-request/psbt-decoded-request-nodes/outputs/psbt-decoded-outputs.tsx

```diff
@@ -0,0 +1,59 @@
+import * as btc from '@scure/btc-signer';
+import { Box, Text } from '@stacks/ui';
+
+import { satToBtc } from '@app/common/money/unit-conversion';
+
+import { PsbtBtcOutput } from '../psbt-btc-output';
+import { PsbtDecodedOutputLayout } from './psbt-decoded-output.layout';
+import { PsbtPlaceholderOutputs } from './psbt-placeholder-outputs';
+
+// TODO: Handle all address types?
+function getAddressFromScript(script: Uint8Array) {
+  const outScript = btc.OutScript.decode(script);
+
+  if (outScript.type === 'wsh') return btc.programToWitness(0, script.slice(2));
+  if (outScript.type === 'wpkh') return btc.programToWitness(0, script.slice(2));
+  if (outScript.type === 'tr') return btc.programToWitness(1, script.slice(2));
+  return '';
+}
+
+function isPlaceholderTransaction(address: string, outputs: btc.TransactionOutputRequired[]) {
+  const outputsNotToCurrentAddress = outputs.filter(output => {
+    const addressFromScript = getAddressFromScript(output.script);
+    return addressFromScript !== address;
+  });
+  return outputsNotToCurrentAddress.length === 0;
+}
+
+interface PsbtDecodedOutputs {
+  address: string;
+  outputs: btc.TransactionOutputRequired[];
+  onSetShowPlaceholder(): void;
+}
+export function PsbtDecodedOutputs({ address, onSetShowPlaceholder, outputs }: PsbtDecodedOutputs) {
+  if (isPlaceholderTransaction(address, outputs)) {
+    onSetShowPlaceholder();
```

Comment:
This is weird, because you're invoking a callback during the render flow. This will call many times. Should be in an effect.

Reply:
Yeah, good point. Will change.